lib/models/message.dart

```diff
@@ -0,0 +1,23 @@
+import 'package:cloud_firestore/cloud_firestore.dart';
+
+class Message {
```

Comment:
I would argue for immutable PODOs as models but I'm not suggesting the PR be held up by a discussion, I just wanted to flag the choice for a discussion that we can have over time.

Reply:
You're right. I'll add a const constructor.

Reply:
Or did you mean using the `@immutable` annotation?

Reply:
Yeah that's partly why I thought it was better as a longer discussion - as I'm sure you know there's no proper immutability at the language level with Dart so the options are (as far as I'm aware): 
- a combination of a bunch of language features (const, annotation, immutable collection types, etc. ... ie. the DIY option) or 
- choose from a 3rd part package (built_value and freezed are the main two I think) 

DIY is clunky at best imo and lots of work so personally (very happy to be voted down) it's a choice between the packages or don't bother with immutability.

Reply:
I understand. I'll leave it as it is for now.

Reply:
The one thing I think could (potentially) be worth discussing at this point is the models being PODOs.  I'm probably not using the right term for what I'm trying to achieve - it's the dependency on the firestore that I was thinking about.  I try and not have any dependencies on higher level modules in my models, I've always thought it was to try and follow SOLID, specifically DIP and maybe a bit of SRP.  

Personally I have been using extension methods to be able to have a .fromX method like you do here but not have a dependency on X in the model.  I'd be happy to branch off your branch and show you what I mean if you like.  The main disadvantage imo is that right now the code completion doesn't work out of the box until you manually import the extension, but there's a fix on the way for the analyzer which sounds like it shouldn't be far away.

Reply:
But I just thought it was a good chance to talk about it, please don't let it hold up the PR unless you feel like it.